docs/advanced/instrumentation.md

```diff
@@ -121,35 +121,183 @@ If we wouldn't do this, the result for an async data fetcher would always be 0,
     }
     ```
 
 
 Datafetcher 'Query.shows': 0ms
 
-Total execution time: 3ms
\ No newline at end of file
+Total execution time: 3ms
+
+
+## Metrics Out of The Box
+
+!!! abstract "tl;dr"
+
+    * Supported vi the opt-in `graphql-dgs-spring-boot-micrometer` module.
+    * Provides specific GraphQL metrics such as `gql.query`, `gql.error`, and `gql.dataLoader`.
+    * Backed by [Micrometer], it supports several backends.
+
+
+=== "Gradle Groovy"
+    ```groovy
+    dependencies {
+        implementation 'com.netflix.graphql.dgs:graphql-dgs-spring-boot-micrometer:3.+'
+    }
+    ```
+=== "Gradle Kotlin"
+    ```kotlin
+    dependencies {
+        implementation("com.netflix.graphql.dgs:graphql-dgs-spring-boot-micrometer:3.+")
+    }
+    ```
+=== "Maven"
+    ```xml
+    <dependencies>
+        <dependency>
+            <groupId>com.netflix.graphql.dgs</groupId>
+            <artifactId>graphql-dgs-spring-boot-micrometer</artifactId>
+            <version>3.+</version>
+        <dependency>
+    </dependencies>
+    ```
+!!! warning
+    The version used above is just an example. Please verify the verison you want to use by visiting
+    the [Release page](https://github.com/Netflix/dgs-framework/releases).
+
+
+### Query Timer: gql.query
+
+Captures the elapsed time that a given GraphQL query, or mutation, takes.
+
+**Name:** `gql.query`
+
+**Tags:**
+
+| tag name     | values                 |description |
+| ------------ | ---------------------- | ---------- |
+| `outcome`    | `success` or `failure` | Result of the operation, as defined by the [ExecutionResult].
+
+
+### Error Counter: gql.error
+
+Captures the number of GraphQL errors encountered during query execution of a query or mutation.
+Remember that one _graphql_ request can have multiple errors.
+
+**Name:** `gql.error`
+
+**Tags:**
+
+| tag name          | description |
+| ----------------- | ----------- |
+| `gql.errorCode`   | The GraphQL error code, such as `VALIDATION`, `INTERNAL`, etc.|
+| `gql.path`        | The sanitized query path that resulted in the error.     |
+| `gql.errorDetail` | Optional flag containing additional details, if present. |
+
+### Data Loader Timer: gql.dataLoader
+
+Captures the elapsed time for a data loader invocation for a batch of queries.
+This is useful if you want to find data loaders that might be responsible for poor query performance.
+
+**Name:** `gql.dataLoader`
+
+**Tags:**
+
+| tag name              | description |
+| --------------------- | ----------- |
+| `gql.loaderName`      | The name of the data loader, may or may not be the same as the type of entity. |
+| `gql.loaderBatchSize` | The number of queries executed in the batch. |
+
+
+### Data Fetcher Timer: gql.resolver
+
+Captures the elapsed time of each data fetcher invocation.
+This is useful if you want to find data fetchers that might be responsible for poor query performance.
+
+!!! note
+    This metric is not available if the data is resolved via a Batch Loader.
```

Comment:
By default this is true. Previously you could override the behavior with an annotation, to explicitly enable/disable metrics on a fetcher. Is that still supported?

Reply:
We don't supported in the OSS module, will open a ticket to add it.

Reply:
Tracking via https://github.com/Netflix/dgs-framework/issues/251